plasmapy/analysis/fit_functions.py

```diff
@@ -0,0 +1,1141 @@
+            dependent variables corresponding to :math:`x`
+
+        """
+        return self._explin.func(x, a, alpha, 0.0, b)
+
+    def func_err(self, x, x_err=None, rety=False):
+        """
+        Calculate dependent variable uncertainties :math:`\\delta y` for
+        dependent variables :math:`y=f(x)`.
+
+        .. math::
+
+            (\\delta y)^2 =
+                \\left( A e^{\\alpha x}\\right)^2 \\left[
+                    \\left( \\frac{\\delta A}{A} \\right)^2
+                    + (x \\, \\delta \\alpha)^2
+                    + (\\alpha \\, \\delta x)^2
+                \\right]
+                + (\\delta b)^2
+
+        Parameters
+        ----------
+        x: array_like
+            Independent variables to be passed to the fit function.
+
+        x_err: array_like, optional
+            Errors associated with the independent variables `x`.  Must be of
+            size one or equal to the size of `x`.
+
+        rety: bool
+            Set `True` to also return the associated dependent variables
+            :math:`y = f(x)`.
+
+        Returns
+        -------
+        err: `numpy.ndarray`
+            The calculated uncertainties :math:`\\delta y` of the dependent
+            variables (:math:`y = f(x)`) of the independent variables `x`.
+
+        y: `numpy.ndarray`, optional
+            (if `rety = True`) The associated dependent variables
+            :math:`y = f(x)`.
+        """
+        y, err = self._explin(x, x_err=x_err, reterr=True)
+
+        if rety:
+            return err, y
+
+        return err
+
+    def root_solve(self, *args, **kwargs):
+        """
+        The root :math:`f(x_r) = 0` for the fit function.
+
+        .. math::
+
+            x_r &= \\frac{1}{\\alpha} \\ln \\left( \\frac{-b}{A} \\right)
+
+            \\delta x_r &= \\sqrt{
+                \\left( \\frac{1}{\\alpha} \\frac{\\delta A}{A} \\right)^2
+                + \\left( x_r \\frac{\\delta \\alpha}{\\alpha} \\right)^2
+                + \\left( \\frac{1}{\\alpha} \\frac{\\delta b}{b} \\right)^2
+            }
+
+        Parameters
+        ----------
+        *args
+            Not needed.  This is to ensure signature comparability with
+            `AbstractFitFunction`.
+
+        *kwargs
+            Not needed.  This is to ensure signature comparability with
+            `AbstractFitFunction`.
+
+        Returns
+        -------
+        root: float
+            The root value for the given fit :attr:`parameters`.
+
+        err: float
+            The uncertainty in the calculated root for the given fit
+            :attr:`parameters` and :attr:`parameters_err`.
+        """
+        a, alpha, b = self.params
+        a_err, b_err, c_err = self.param_errors
+
+        root = np.log(-b / a) / alpha
+
+        a_term = a_err / (a * alpha)
+        b_term = b_err * root / alpha
+        c_term = c_err / (alpha * b)
+        err = np.sqrt(a_term ** 2 + b_term ** 2 + c_term ** 2)
+
+        return root, err
```

Comment:
```suggestion
        return root, err
```
Again, perhaps a namedtuple?